src/python/pants/option/option_types.py

```diff
@@ -97,7 +97,7 @@ def __new__(
         removal_version: str | None = None,
         removal_hint: str | None = None,
         deprecation_start_version: str | None = None,
-        environment_sensitive: bool = False,
+        environment_sensitive: bool | None = None,
```

Comment:
Why is this tri-value now?

Reply:
@stuhood I realised from [precedent from `advanced`](https://github.com/pantsbuild/pants/blob/0b8e4f2317f0157648500d1db553a7a0a21607f9/src/python/pants/option/option_types.py#L92), that values are non-None only for options that have been explicitly registered. Having `environment_sensitive` always be set was causing tests to fail in `option_types_test`, and it didn't make sense to always call out `environment_sensitive` but not e.g. `advanced`.

Reply:
Note that in practice, `None` is treated as Falsey for `advanced` and behaviour is similar for `environment_sensitive` here.